translator/c/python/op2.py

```diff
@@ -129,6 +134,10 @@ def self_evaluate_macro_defs(macro_defs):
           # print("Performing a substitution of '" + k2 + "'->'" + k2_val + "' into '" + k_val + "' to produce '" + macro_defs[k] + "'")
           substitutions_performed = True
 
+          num_subs_performed = num_subs_performed + 1
+          if num_subs_performed == max_subs_allowed:
```

Comment:
Could there be cases where this limit is reached *without* it being an infinite loop ?

Reply:
Hypothetically, yes. If resolving all macro definitions across a codebase required 10,000 substitutions then that would halt the translator.

Reply:
I can change the response to a detected infinte loop, so instead of terminating the translator I instead just stop the macro substitution. Is this reasonable to you?

Reply:
I wonder if there is a way to figure out how many macro-substitutions will exist in total for a code base and then use that as the max_subs_allowed. This way the upper limit is not arbitrarily set and depends on the code base.

Reply:
I understand. I believe that this can be done, and precisely too.

Reply:
See latest commit.